src/vim9expr.c

```diff
@@ -439,6 +459,12 @@ compile_class_object_index(cctx_T *cctx, char_u **arg, type_T *type)
 	    ocmember_T *m = &cl->class_class_members[idx];
 	    if (STRNCMP(name, m->ocm_name, len) == 0 && m->ocm_name[len] == NUL)
 	    {
+		if ((cl->class_flags & CLASS_INTERFACE) != 0)
+					//&& m->ocm_type->tt_type == VAR_CLASS)
+		{
+		    semsg(_(e_interface_static_direct_access_str), m->ocm_name);
+		    return FAIL;
```

Comment:
Can you add tests for the two new cases?

Reply:
> Can you add tests for the two new cases?

Of course, when it's ready to come out of draft. 

I've left this in draft because of the following problem, mentioned above in https://github.com/vim/vim/pull/13007#issuecomment-1703677466. In vim9expr.c I left a chunk of _not working_ code commented out. I'm still investigating; do you have any suggestions , or can point me at an example that's similar?

What about the following which fails with
```
Error detected while compiling :source buffer=1[14]..function <SNR>36_F:
line    1:
E1326: Member not found on object "I": s_var
```
produce by
```
vim9script

interface I
    public static s_var: number
endinterface
class C implements I
    public static s_var = 3
endclass

def F(i: I)
    var x = i.s_var
enddef
var c = C.new()
F(c)
```

Reply:
> Can you add tests for the two new cases?

See `Test_interface_static_member_access()`